.globalconfig

```diff
@@ -0,0 +1,4 @@
+﻿is_global = true
+
+# IDE0059: Remove unnecessary value assignment
+dotnet_diagnostic.IDE0059.severity = warning
```

Comment:
I'm curious, why a global config, and not just add it to our existing editor config? I've never really understood how these work :)

Reply:
I also don't fully understand; but I read a bit of [the docs](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/configuration-files) and my tl;dr is that editorconfigs can be filtered to just some files; globalconfigs apply to all code. Seemed like we want to apply this to all code. But it also feels weird to create a new file, maybe I will move it to the editorconfig (if it's under `[*.cs]`, the two approaches should be mostly equivalent I guess), you're right.

But also I really wanted to enable this for the Compiler layer, but then I found we have this:

https://github.com/dotnet/razor/blob/62c5f524fb221efc1bcff04c03352a8ac705cd7b/src/Compiler/Directory.Build.props#L17

so I need to turn that on and fix other warnings that will appear as result of that...

Reply:
I guess the "merge the .editorconfigs" never happened after the merge, because it looks like compiler has its own, which is fine, but its marked as `root` so the one in the folder above it won't be picked up: https://github.com/dotnet/razor/blob/62c5f524fb221efc1bcff04c03352a8ac705cd7b/src/Compiler/.editorconfig#L14

So that means either add it to both `.editorconfig` files, or keep this approach.

Reply:
I guess we can merge this PR and I will follow up with enabling it in the compiler layer in a separate PR, since y'all already reviewed this one.

Reply:
That makes sense to me. I wouldn't block this PR on getting our `.editorconfig` story clean.